components/playlist/resources/playerApiSink.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { getAllActions } from './api/getAllActions'
+import { PlayerMessagePayload } from './api/playerApi'
+
+import { types } from './constants/playlist_types'
+
+export default function startReceiving () {
+  window.onmessage = (e) => {
```

Comment:
please abort if the message origin is not expected, like in this example: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#the_dispatched_event

Reply:
Done. the `Security concerns` section is so helpful. Thanks.